File: learning/KadanAlgorithm.py

# def kadan_algo(arr):
#     cs=0
#     ms = float('-inf')
#     for num in arr:
#         cs=cs+num
#         if cs<0:
#             cs=0
#         ms =max(ms,cs)
#     return ms


# print(kadan_algo([-2,-3,4,-1,-2,1,5,-2]))

def kadan_algo(arr):
    ms = float('-inf')
    cs = 0
    for num in arr:
        cs = cs+num
        if cs<0:
            cs=0
        ms=max(ms,cs)
    return ms

arr=[-2,-3,4,-1,-2,1,5,-3]
print(kadan_algo(arr))

File: learning/KadanAlgorithm.java
public class KadanAlgorithm {
    public static void main(String[] args) {
        KadanAlgo ka = new KadanAlgo();
        int arr[] = { -2, -3, 4, -1, -2,1, 5, -3 };
        System.out.println(ka.applyKadanAlgo(arr));
    }
}
class KadanAlgo{
    public int applyKadanAlgo(int[] arr) {
        int cs = 0;
        int ms = Integer.MIN_VALUE;
        for (int i = 0; i < arr.length; i++) {
            cs = cs + arr[i];
            if (cs < 0) {
                cs = 0;
            }

            ms = Math.max(cs, ms);
            
        }
        return ms;
    }
}